### PR TITLE
Fix flaky test for #2293

### LINKED ISSUE #2360: intermittent failure of ztests/fixedbugs/2241.yaml
This test fails intermittently...
```
ztest.go:475: /home/runner/work/zq/zq/ztests/fixedbugs/2241.yaml: expected and actual stdout differ:
        --- expected
        +++ actual
        @@ -1 +1,2 @@
        +added: file:///tmp/2241719424260/zartest/zd/20180324/d-1ps7DSsgnER2VwIUuWDHvPWRIOU.zng.zar/idx-1ps7DNhgHIEZfWSR83DxdJ4wEJj.zng
```
It looks like there is output status racing against the zar command exiting.  Seems like zar should wait for all such output?

### PR DESCRIPTION
**Pull request by mattnibs**
The ztest for fixedbugs/2293 should have used the "-q" flag since the test does
not care about the output of "zar index create".

Never the less, the flaky test exposed an race conditition with the progress
update functionality, where the program would exit before the update logic
would have a chance to print all updates. Fix this as well.

Also the test was named after the wrong issue, update the name to reflect the
right issue.

Closes #2360

### CODE REVIEW

**Review by mccanne — APPROVED** [review]
Yay!

File: ppl/cmd/zar/index/create.go [hunk 1]
```diff
@@ -43,29 +44,32 @@ requires specifying the key and output file name. For example:
 
 type CreateCommand struct {
 	*root.Command
+	displayer  *displayer
```
Comment (nwt) [inline ppl/cmd/zar/index/create.go:47]: Make this `displayer displayer` so you don't have to initialize it?

File: ppl/cmd/zar/index/create.go [hunk 2]
```diff
@@ -43,29 +44,32 @@ requires specifying the key and output file name. For example:
...
 	framesize  int
 	inputFile  string
 	keys       string
 	outputFile string
 	ensure     bool
 	noapply    bool
 	procFlags  procflags.Flags
-	progress   chan string
 	quiet      bool
```
Comment (nwt) [inline ppl/cmd/zar/index/create.go:55]: Is `quiet` now unused?

File: ppl/cmd/zar/index/create.go [hunk 3]
```diff
@@ -110,14 +114,13 @@ func (c *CreateCommand) Run(args []string) error {
...
-	}
+	c.displayer.run()
+	defer c.displayer.close()
 
-	if err := lake.WriteIndices(ctx, lk, c.progress, defs...); err != nil {
+	if err := lake.WriteIndices(ctx, lk, c.displayer.ch, defs...); err != nil {
 		return err
 	}
+
 	return nil
```
Comment (nwt) [inline ppl/cmd/zar/index/create.go:124]: Shorten to `return lake.WriteIndices(ctx, lk, c.displayer.ch, defs...)`?

File: ppl/cmd/zar/index/create.go (hunk 1 above)
Comment (mattnibs) [reply ppl/cmd/zar/index/create.go:47]: yup

File: ppl/cmd/zar/index/create.go (hunk 2 above)
Comment (mattnibs) [reply ppl/cmd/zar/index/create.go:55]: correct will remove

**Pull request merged**